src/app/singletons/fighter_manager.rs

```diff
@@ -234,33 +234,33 @@ impl FighterManager {
         }
     }
 
-    pub fn get_fighter_entry<'a>(&'a self, entry_id: app::FighterEntryID) -> &'a app::FighterEntry {
+    pub fn get_fighter_entry<'a>(&'a self, entry_id: app::FighterEntryID) -> Option<&'a app::FighterEntry> {
         unsafe {
-            &*impl_::get_fighter_entry(self, entry_id)
+            impl_::get_fighter_entry(self, entry_id).as_ref()
         }
     }
 
-    pub fn get_fighter_information<'a>(&'a self, entry_id: app::FighterEntryID) -> &'a app::FighterInformation {
+    pub fn get_fighter_information<'a>(&'a self, entry_id: app::FighterEntryID) -> Option<&'a app::FighterInformation> {
         unsafe {
-            &*impl_::get_fighter_information(self, entry_id)
+            impl_::get_fighter_information(self, entry_id).as_ref()
         }
     }
 
-    pub fn get_fighter_entry_mut<'a>(&'a mut self, entry_id: app::FighterEntryID) -> &'a mut app::FighterEntry {
+    pub fn get_fighter_entry_mut<'a>(&'a mut self, entry_id: app::FighterEntryID) -> Option<&'a mut app::FighterEntry> {
         unsafe {
-            &mut *impl_::get_fighter_entry(self, entry_id)
+            impl_::get_fighter_entry(self, entry_id).as_mut()
         }
     }
 
-    pub fn get_fighter_information_mut<'a>(&'a mut self, entry_id: app::FighterEntryID) -> &'a mut app::FighterInformation {
+    pub fn get_fighter_information_mut<'a>(&'a mut self, entry_id: app::FighterEntryID) -> Option<&'a mut app::FighterInformation> {
```

Comment:
I know that I originally wrote this with the generic but if you can check if it's not needed anymore? Thanks